### PR TITLE
wal-lang: init at rev 06498c3f5341ce687a37ad52647f300b72dff52a

### PR DESCRIPTION
**Pull request by Avimitin**
## Description of changes

This PR adds three new packages [wal-lang](https://github.com/ics-jku/wal), lark-parser and pylibfst. lark-parser and pylibfst are required by wal-lang, but not presented in python-modules. And we need to use the HEAD revision because the latest release of the wal-lang repository is missing features and has some of the dependencies with security issues.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/lark-parser/default.nix
Comment (MikaelFangel) [inline pkgs/development/python-modules/lark-parser/default.nix:15]: There should be listed a maintainer.

File: pkgs/development/python-modules/lark-parser/default.nix
Comment (MikaelFangel) [inline pkgs/development/python-modules/lark-parser/default.nix:14]:
```suggestion
    description = "A modern general-purpose parsing library";
```

File: pkgs/development/python-modules/lark-parser/default.nix
Comment (MikaelFangel) [inline pkgs/development/python-modules/lark-parser/default.nix:16]:
```suggestion
    license = licenses.mit;
```

File: pkgs/development/python-modules/pylibfst/default.nix
Comment (MikaelFangel) [inline pkgs/development/python-modules/pylibfst/default.nix:28]:
```suggestion
    license = licenses.bsd3;
```

File: pkgs/development/python-modules/pylibfst/default.nix
Comment (MikaelFangel) [inline pkgs/development/python-modules/pylibfst/default.nix:27]: There should be listed a maintainer.

File: pkgs/development/python-modules/pylibfst/default.nix
Comment (MikaelFangel) [inline pkgs/development/python-modules/pylibfst/default.nix:5]:
```suggestion
  format = "pyproject";

```

File: pkgs/development/python-modules/wal-lang/default.nix
Comment (MikaelFangel) [inline pkgs/development/python-modules/wal-lang/default.nix:36]:
```suggestion
    license = licenses.bsd3;
```

File: pkgs/development/python-modules/wal-lang/default.nix
Comment (MikaelFangel) [inline pkgs/development/python-modules/wal-lang/default.nix:37]: Again here should be a maintainer

File: pkgs/development/python-modules/wal-lang/default.nix
Comment (MikaelFangel) [inline pkgs/development/python-modules/wal-lang/default.nix:35]:
```suggestion
    description = "Programmable waveform analysis";
```

**Review by MikaelFangel — CHANGES_REQUESTED** [review]

**Comment by MikaelFangel** [comment]
The commit adding you as a maintainer should be the first commit of this PR with the format:
`maintainers: add <handle>`

**Comment by Avimitin** [comment]
Should be fixed. Thanks for your review.

**Comment by Avimitin** [comment]
Seems like that I can only change the base branch. Is it ok to merge this PR? Or should I open a new one?

**Comment by MikaelFangel** [comment]
A member with the right privileges needs to approve the checks and only a member with merge privileges (not me) can merge this PR. Besides I think other more experienced members also should review this so it follows all the guidelines (:

**Comment by Avimitin** [comment]
Ok I get that. Still, thanks for your review.

File: pkgs/development/python-modules/lark-parser/default.nix
Comment (leo60228) [inline pkgs/development/python-modules/lark-parser/default.nix:2]: There's an extra space at the beginning of this line.

**Review by leo60228 — CHANGES_REQUESTED** [review]
This should probably be split into multiple commits, and I think the commit message should be formatted like "wal-lang: init at unstable-2023-08-12".

Additionally, while I don't know much about the project being packaged and using an unstable version might be standard, would it be possible to use a stable release with the dependency updates backported? I assume there's a reason that the new features aren't in a release yet, and Nixpkgs generally packages stable versions when possible.

**Comment by Avimitin** [comment]
> would it be possible to use a stable release with the dependency updates backported?

Umm that means we have to maintain a patch for this. Using latest commit will be more convenient. But actually it is stable enough, maybe I should go to upstream and ask them to release a new version.

**Comment by Avimitin** [comment]
Seems like the No channel PR action cannot pass successfully. Maybe I should open a new branch and new PR for this package?

**Comment by Avimitin** [comment]
Closed the PR for now. I will try to argue with upstream to get new release.

**Pull request closed without merging**
